Documentation/contributing/testing/e2e_legacy.rst

```diff
@@ -0,0 +1,739 @@
+      It("Example test", func(){
+          Expect(true).Should(BeTrue())
+      })
+
+      FIt("Example focused test", func(){
+          Expect(true).Should(BeTrue())
+      })
+
+
+* From the command line: specify a more granular focus if you want to focus on, say, Runtime L7 tests:
+
+  .. code-block:: shell-session
+
+      INTEGRATION_TESTS=true ginkgo --focus "Runtime.*L7"
+
+
+This will focus on tests that contain "Runtime", followed by any
+number of any characters, followed by "L7". ``--focus`` is a regular
+expression and quotes are required if it contains spaces and to escape
+shell expansion of ``*``.
+
+.. _Focused Specs: https://onsi.github.io/ginkgo/#focused-specs
+
+Compiling the tests without running them
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To validate that the Go code you've written for testing is correct without
+needing to run the full test, you can build the test directory:
+
+.. code-block:: shell-session
+
+    make -C test/ build
+
+Updating Cilium images for Kubernetes tests
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Sometimes when running the CI suite for a feature under development, it's common
+to re-run the CI suite on the CI VMs running on a local development machine after
+applying some changes to Cilium. For this the new Cilium images have to be
+built, and then used by the CI suite. To do so, one can run the following
+commands on the ``k8s1`` VM:
+
+.. code-block:: shell-session
+
+   cd go/src/github.com/cilium/cilium
+
+   make LOCKDEBUG=1 docker-cilium-image
+   docker tag quay.io/cilium/cilium:latest \
+	k8s1:5000/cilium/cilium-dev:latest
+   docker push k8s1:5000/cilium/cilium-dev:latest
+
+   make -B LOCKDEBUG=1 docker-operator-generic-image
+   docker tag quay.io/cilium/operator-generic:latest \
+	k8s1:5000/cilium/operator-generic:latest
+   docker push k8s1:5000/cilium/operator-generic:latest
+
+The commands were adapted from the ``test/provision/compile.sh`` script.
+
+Test Reports
+~~~~~~~~~~~~
+
+The Cilium Ginkgo framework formulates JUnit reports for each test. The
+following files currently are generated depending upon the test suite that is ran:
+
+* runtime.xml
+* K8s.xml
+
+Best Practices for Writing Tests
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* Provide informative output to console during a test using the `By construct <https://onsi.github.io/ginkgo/#documenting-complex-its-by>`_. This helps with debugging and gives those who did not write the test a good idea of what is going on. The lower the barrier of entry is for understanding tests, the better our tests will be!
+* Leave the testing environment in the same state that it was in when the test started by deleting resources, resetting configuration, etc.
+* Gather logs in the case that a test fails. If a test fails while running on Jenkins, a postmortem needs to be done to analyze why. So, dumping logs to a location where Jenkins can pick them up is of the highest imperative. Use the following code in an ``AfterFailed`` method:
```

Comment:
```suggestion
* Gather logs in case a test fails. If a test fails while running on Jenkins, do a postmortem to analyze why. It's important to dump logs to a location where Jenkins can pick them up. Use the following code in an ``AfterFailed`` method:
```